tests/test_lanczos_device.cu

```diff
@@ -0,0 +1,118 @@
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * This file is part of the mfmg libary. mfmg is distributed under a BSD *
+ * 3-clause license. For the licensing terms see the LICENSE file in the *
+ * top-level directory                                                   *
+ *                                                                       *
+ * SPDX-License-Identifier: BSD-3-Clause                                 *
+ *************************************************************************/
+
+#define BOOST_TEST_MODULE lanczos_device
+
+#include <mfmg/common/lanczos.templates.hpp>
+#include <mfmg/cuda/utils.cuh>
+
+#include <boost/test/data/test_case.hpp>
+
+#include <cmath>
+#include <cstdio>
+
+#include "lanczos_simpleop.templates.hpp"
+#include "main.cc"
+
+namespace bdata = boost::unit_test::data;
+namespace tt = boost::test_tools;
+
+BOOST_DATA_TEST_CASE(lanczos,
+                     bdata::make({1, 2}) * bdata::make({false, true}) *
+                         bdata::make({1, 2, 3, 5}),
+                     multiplicity, is_deflated, n_distinct_eigenvalues)
+{
+  using namespace mfmg;
+  std::cout << multiplicity << " " << is_deflated << " "
+            << n_distinct_eigenvalues << std::endl;
+
+  using VectorType =
+      dealii::LinearAlgebra::distributed::Vector<double,
+                                                 dealii::MemorySpace::CUDA>;
+  using OperatorType = SimpleOperator<VectorType>;
+
+  const int n = 1000;
+  const int n_eigenvectors = n_distinct_eigenvalues * multiplicity;
+
+  OperatorType op(n, multiplicity);
+
+  boost::property_tree::ptree lanczos_params;
+  lanczos_params.put("is_deflated", is_deflated);
+  lanczos_params.put("num_eigenpairs", n_eigenvectors);
+  if (is_deflated)
+  {
+    lanczos_params.put("num_cycles", multiplicity);
+    lanczos_params.put("num_eigenpairs_per_cycle", n_distinct_eigenvalues);
+  }
+  else
+  {
+    if (multiplicity > 1)
+    {
+      // Skip test when using regular Lanczos for multiplicities larger than 1
+      return;
+    }
+  }
+
+  lanczos_params.put("max_iterations", 200);
+  lanczos_params.put("tolerance", 1e-2);
+  lanczos_params.put("percent_overshoot", 5);
+
+  Lanczos<OperatorType, VectorType> solver(op);
+
+  dealii::LinearAlgebra::distributed::Vector<double> initial_guess_host(n);
+  initial_guess_host = 1.;
+
+  // Add random noise to the guess
+  std::mt19937 gen(0);
+  std::uniform_real_distribution<double> dist(0, 1);
+  std::transform(initial_guess_host.begin(), initial_guess_host.end(),
+                 initial_guess_host.begin(),
+                 [&](auto &v) { return v + dist(gen); });
+  auto initial_guess_dev = copy_from_host(initial_guess_host);
+  std::cout << "initial guess " << initial_guess_dev.l2_norm() << " "
+            << initial_guess_host.l2_norm() << std::endl;
```

Comment:
Is this output intentionally here?

(Comparing with the host version it is not there.)